lib/charms/layer/jenkins/packages.py

```diff
@@ -92,7 +92,8 @@ def _setup_source(self, release):
             raise Exception(message)
 
         # Setup archive to use appropriate jenkins upstream
-        wget = ("wget", "-q", "-O", "-", APT_KEY % dist)
         source = APT_SOURCE % dist
-        key = subprocess.check_output(wget).decode("utf-8")
-        self._apt.add_source(source, key=key)
+        keyfile = os.path.join(hookenv.charm_dir(), "jenkins.io.key")
+        with open(keyfile, "r") as k:
+            key = k.read()
+        self._apt.add_source(source, key = key)
```

Comment:
As per per [PEP8](https://www.python.org/dev/peps/pep-0008/#other-recommendations), please avoid spaces in keyword arguments:

```python
self._apt.add_source(source, key=key)
```